### DISCUSSION TITLE
No suggestions for creating anything other than Interface/Class

### DISCUSSION THREAD

**Discussion by TheColorRed** [cyrilletuzi/vscode-angular-schematics#925] [Q&A] [answered]
# Checks before posting an issue

- [x] I have read _all_ the documentation included in the extension itself as a walkthrough, including the [troubleshooting guide](https://github.com/cyrilletuzi/vscode-angular-schematics/blob/main/TROUBLESHOOTING.md) (from the VS Code menu: View > Command Palette >search "Documentation" > choose "Angular Schematics: Documentation and tutorial")
- [x] I have checked in already:
  - [opened issues](https://github.com/cyrilletuzi/vscode-angular-schematics/issues)
  - [closed issues](https://github.com/cyrilletuzi/vscode-angular-schematics/issues?q=is%3Aissue+is%3Aclosed)
  - [questions answered](https://github.com/cyrilletuzi/vscode-angular-schematics/discussions/categories/q-a) in the new "Discussions" tab
- [x] My project meets the requirements indicated in README
- [x] My workspace is in [trusted mode](https://code.visualstudio.com/docs/editor/workspace-trust), not restricted mode
- [x] I looked at the extension's logs ("Angular Schematics" Output channel, second tab left to the Terminal), and tried to fix my project's configuration accordingly
- [x] I understand I am writing to a *human being*
- [x] I will be involved in my own issue, by answering questions and informing if the issue is resolved

# Configuration

- VS Code version (only the last 2 versions are supported): Version: 1.87.0 (Universal)
- Extension version: v6.9.0
- OS: MacOS
- Are you using VS Code in normal mode or in a special mode (web version, WSL, SSH/remote, etc.): Normal

# Description of the issue

I am using [NX](https://nx.dev/) for my work project, and a few releases ago they removed the need for an `angular.json`, I am not sure if this is causing an issue for creating a schematic or not, but on my personal computer, I am able to create schematics without NX.

Each library in a NX file has a `project.json` instead of an `angular.json`. The `project.json` is a combination of nx settings and `angular.json` settings.

![image](https://github.com/cyrilletuzi/vscode-angular-schematics/assets/1571806/9120ba7b-e7a8-4611-8440-6901a0246997)

# Extension logs

```
[Info - 8:53:30 AM] Starting journey to generate a schematics.
[Info - 8:53:30 AM] Generation started from a file/folder selection: /Users/rnaddy/Documents/vscode/netra/angular-ui-component-library/netra
[Info - 8:53:30 AM] User has selected a file/folder: resolving current workspace folder from it.
[Info - 8:53:30 AM] Workspace folder selected: "angular-ui-component-library"
```

# How to reproduce the issue

1. Right click on a folder
2. Select `Angular Schematics: Generate a file`

**Comment by cyrilletuzi** [comment, marked as answer]
Hello,

As explained in the troubleshooting section of the documentation, displayed schematics are based on your `package.json` dependencies.

The fact that 2 TypeScript schematics are proposed means that at least one `package.json` has been found, with `typescript` as a dependency, but that no `@angular/core` dependency was found in any `package.json`.

So I would need more information about your `package.json` to be able to help you.

As a workaround, you can use the extension's configuration helper to force dependencies.
